### PR TITLE
Add class for showing pop-up warnings to user

### PR DESCRIPTION
**Pull request by AlexejheroYTB**

### CODE REVIEW

File: QModManager/ErrorMessage.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+namespace QModManager
+{
+    public static class ErrorMessage
+    {
+        public static void ShowError(string error, Action<bool> function, string leftButtonText = "Yes", string rightButtonText = "No")
+        {
+            // Still need to implement leftButtonText and rightButtonText
+            uGUI_SceneConfirmation confirmation = uGUI.main.confirmation;
+
+            if (leftButtonText == "" || leftButtonText == null) confirmation.no.gameObject.SetActive(false);
```
Comment (PrimeSonic) [inline QModManager/ErrorMessage.cs]: Use `string.IsNullOrEmpty(` when `string` values for null and empty.

Comment (AlexejheroYTB) [reply QModManager/ErrorMessage.cs]: Addressed in commit 837a2bd

**Comment by AlexejheroYTB** [comment]
Still need to figure out how to change the text - It's hard when you don't have a scene debugger

**Comment by ahk1221** [comment]
I'll take a look.

**Comment by AlexejheroYTB** [comment]
There is an issue with the buttons not working properly - this pr will be put on hold until #26 is fixed

File: QModManager/Error.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using UnityEngine.UI;
+
+namespace QModManager
+{
+    internal static class Error
+    {
+        internal static void ShowError(string error, Action onLeftButtonPressed = null, Action onRightButtonPressed = null, string leftButtonText = "See Log", string rightButtonText = "Close")
```
Comment (PrimeSonic) [inline QModManager/Error.cs]: Optional parameters of the same type can make for messy calls of the method in question.  
You may often find that you will have to constantly add the parameter names to make this work.

**Proposal:**  
Create a new abstract class `ButtonAction` that exposes a `string Text` property and a `Action ButtonPressed` property.  
Then, you can create `LeftButtonAction : ButtonAction` and `RightButtonAction : ButtonAction` classes and use them as optional parameters here with `null` as their default.  
```CSharp
ShowError(string error, LeftButtonAction leftButton = null, RightButtonAction rightButton = null);
```
This way, there is no ambiguity in the call and no need to use parameter names to clearly express which text and which action belong to which button.

File: QModManager/Error.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            else confirmation.yes.gameObject.GetComponentInChildren<Text>().text = leftButtonText;
+            if (string.IsNullOrEmpty(rightButtonText)) confirmation.no.gameObject.SetActive(false);
+            else confirmation.no.gameObject.GetComponentInChildren<Text>().text = rightButtonText;
+
+            confirmation.Show(error, delegate (bool leftButtonClicked)
+            {
+                if (leftButtonClicked) onLeftButtonPressed.Invoke();
+                else onRightButtonPressed.Invoke();
+
+                Console.Write("");
```
Comment (PrimeSonic) [inline QModManager/Error.cs]: If there is a reason for this, at least say so in a comment.

File: QModManager/Error.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using UnityEngine.UI;
+
+namespace QModManager
+{
+    internal static class Error
```
Comment (PrimeSonic) [inline QModManager/Error.cs]: Best to choose a more descriptive class name. This one could cause confusion.  
Suggestion: `internal static class ErrorDialogs`

File: QModManager/Error.cs (hunk 3 above)
Comment (AlexejheroYTB) [reply QModManager/Error.cs]: The buttons have a stupid tendency to not work 90% of the time. I made some changes that broke them but I wasn't what caused it. This line is a leftover from testing. I will remove it

File: QModManager/Error.cs (hunk 2 above)
Comment (AlexejheroYTB) [reply QModManager/Error.cs]: I personally don't see a point in bundling two parameters in a class, as 3 out of the 4 optional parameters will never be used in the release buld - only for testing

File: QModManager/Error.cs (hunk 3 above)
Comment (AlexejheroYTB) [reply QModManager/Error.cs]: Addressed in commit 10ef4b8

File: QModManager/ErrorDialog.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+using System.Diagnostics;
+using System.IO;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace QModManager
+{
+    internal static class Dialog
+    {
+        internal static void Show(string error, Action onLeftButtonPressed = null, Action onRightButtonPressed = null, string leftButtonText = "See Log", string rightButtonText = "Close", bool blue = false)
```
Comment (ahk1221) [inline QModManager/ErrorDialog.cs]: Maybe have the text parameters before the Action parameters.

Comment (AlexejheroYTB) [reply QModManager/ErrorDialog.cs]: I don't think that this is necessary as there will always be `Action` parameters when there are the `string` parameters. Also, again, most of them won't be used in the final build. And if they will, they'll be named.

Comment (ahk1221) [reply QModManager/ErrorDialog.cs]: Guess it makes sense.

**Pull request merged**